pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1762,8 +1762,20 @@ func (l *LibvirtDomainManager) FreezeVMI(vmi *v1.VirtualMachineInstance, unfreez
 			return err
 		}
 	}
-	_, err = l.virConn.QemuAgentCommand(`{"execute":"guest-fsfreeze-freeze"}`, domainName)
+
+	domain, err := l.virConn.LookupDomainByName(domainName)
+	if err != nil {
+		log.Log.Errorf("Domain lookup failed: %v", err)
+		return err
+	}
+	defer domain.Free()
+
+	mounts, err := getMountPoints(domain)
```

Comment:
@ShellyKa13 Are you okay with the fact that we're doing here 2 api requests to get the domain guest info, and then the actual freeze?

Reply:
I dont think we need to go over the mountpoints, as described in : https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainFSFreeze
If given empty mount point all the mount points will be frozen as it was with calling qemu directly.